python/seldon_core/batch_processor.py

```diff
@@ -316,23 +320,102 @@ def _send_batch_predict_multi_request(
     """
 
     indexes = [x[0] for x in input_data]
-    instance_ids = [x[1] for x in input_data]
+
+    seldon_puid = input_data[0][1]
+    instance_ids = [f"{seldon_puid}-item-{n}" for n, _ in enumerate(input_data)]
+    loaded_data = [json.loads(data[2]) for data in input_data]
 
     predict_kwargs = {}
     tags = {
         "batch_id": batch_id,
     }
-
     predict_kwargs["meta"] = tags
-    predict_kwargs["headers"] = {"Seldon-Puid": instance_ids[0]}
+    predict_kwargs["headers"] = {"Seldon-Puid": seldon_puid}
 
     try:
-        # Initialise concatenated array for data
-        loaded = [json.loads(raw_data[2]) for raw_data in input_data]
-        concat = np.concatenate(loaded)
-        predict_kwargs["data"] = concat
+        # Process raw input format
+        if data_type == "raw":
+            input_tags = [d.get("meta", {}).get("tags", {}) for d in loaded_data]
+            first_input = loaded_data[0]
+
+            # Raw input format in mini-batch mode only work for "data" format
+            if "data" not in first_input:
+                raise ValueError(
+                    "raw input with predict in mini-batch mode requires data payload"
+                )
+            # If-block for ndarray case
+            elif "ndarray" in first_input["data"]:
+                payload_type = "ndarray"
+                names_list = [d["data"]["names"] for d in loaded_data]
+                arrays = [np.array(d["data"]["ndarray"]) for d in loaded_data]
+                if not all_equal(names_list):
```

Comment:
Is this all_equal function impl above a more efficient way, or why not just a simple list comprehension?

Reply:
Seems quite an obscure method - so why not just a simple list comprehsnion?

Reply:
Not sure how list comprehension would help? 
Alternative would be
```
names = names_list[0]
for other_name in names_list[1:]:
   assert other_name == names
```
(or similar)

Reply:
done